packages/@glimmer/runtime/test/initial-render-test.ts

```diff
@@ -54,6 +54,22 @@ class AbstractRehydrationTests extends InitialRenderSuite {
 
 class Rehydration extends AbstractRehydrationTests {
 
+  @test "rehydrates into element with pre-existing content"() {
+    let rootElement = this.delegate.serverEnv.getAppendOperations().createElement('div') as HTMLDivElement;
+    let extraContent = this.delegate.serverEnv.getAppendOperations().createElement('noscript') as HTMLElement;
+    rootElement.appendChild(extraContent);
+
+    let noScriptString = '<noscript></noscript>';
+    let template = '<div>Hi!</div>';
+    this.renderServerSide(template, {}, rootElement);
+    let output = expect(this.serverOutput, 'must renderServerSide before calling assertServerOutput');
+    equalTokens(output, content([noScriptString, OPEN, ...template, CLOSE]));
```

Comment:
This was a little wonky.  It expected `OPEN` and `CLOSE` to wrap the entire content passed to it.  But in this case we are explicitly generating a template whose `OPEN` and `CLOSE` wrap only the template which is injected into a node with stuff in it already.  In this case a noscript tag.